hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ReplicationCommandCreator.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hdds.scm.container.replication;
+
+import org.apache.hadoop.hdds.protocol.DatanodeDetails;
+import org.apache.hadoop.hdds.scm.container.ContainerReplica;
+import org.apache.hadoop.ozone.protocol.commands.SCMCommand;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This interface to create respective commands after processing the replicas
+ * with pending ops and health check results.
+ */
+public interface ReplicationCommandCreator {
```

Comment:
I wonder if a better name would be `ContainerUnderReplicationHandler` or just `UnderReplicationHandler`. Then we could have `ECUnderReplicationHandler`, `RatisUnderReplicationHandler`, etc

Reply:
I was trying to be more expressive here. however the classes are doing more than command creation. So, I changed them to Handler classes. ( To be honest I was little hesitant to keep handler in name as we have many all around :-) )